typescript/cli/src/utils/input.ts

```diff
@@ -53,3 +71,402 @@ export async function inputWithInfo({
+    choices: ReadonlyArray<SearchableCheckboxChoice<Value>>,
+  ) => boolean | string | Promise<string | boolean>;
+  theme?: PartialDeep<Theme<SearchableCheckboxTheme>>;
+};
+
+type Item<Value> = NormalizedChoice<Value> | Separator;
+
+type SearchableCheckboxState<Value> = {
+  options: Item<Value>[];
+  currentOptionState: Record<string, NormalizedChoice<Value>>;
+};
+
+function isSelectable<Value>(
+  item: Item<Value>,
+): item is NormalizedChoice<Value> {
+  return !Separator.isSeparator(item) && !item.disabled;
+}
+
+function isChecked<Value>(item: Item<Value>): item is NormalizedChoice<Value> {
+  return isSelectable(item) && Boolean(item.checked);
+}
+
+function toggle<Value>(item: Item<Value>): Item<Value> {
+  return isSelectable(item) ? { ...item, checked: !item.checked } : item;
+}
+
+function normalizeChoices<Value>(
+  choices: ReadonlyArray<SearchableCheckboxChoice<Value>>,
+): NormalizedChoice<Value>[] {
+  return choices.map((choice) => {
+    const name = choice.name ?? String(choice.value);
+    return {
+      value: choice.value,
+      name,
+      short: choice.short ?? name,
+      description: choice.description,
+      disabled: choice.disabled ?? false,
+      checked: choice.checked ?? false,
+    };
+  });
+}
+
+function sortNormalizedItems<Value>(
+  a: NormalizedChoice<Value>,
+  b: NormalizedChoice<Value>,
+): number {
+  if (a.name > b.name) {
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
+function organizeItems<Value>(
+  items: Array<Item<Value>>,
+  selectableOptionsSeparator?: Separator,
+): Array<Item<Value> | Separator> {
+  const orderedItems = [];
+
+  const checkedItems = items.filter(
+    (item) => !Separator.isSeparator(item) && item.checked,
+  ) as NormalizedChoice<Value>[];
+
+  if (checkedItems.length !== 0) {
+    orderedItems.push(new Separator('--Selected Options--'));
+
+    orderedItems.push(...checkedItems.sort(sortNormalizedItems));
+  }
+
+  orderedItems.push(
+    selectableOptionsSeparator ?? new Separator('--Available Options--'),
+  );
+
+  const nonCheckedItems = items.filter(
+    (item) => !Separator.isSeparator(item) && !item.checked,
+  ) as NormalizedChoice<Value>[];
+
+  orderedItems.push(...nonCheckedItems.sort(sortNormalizedItems));
+
+  if (orderedItems.length === 1) {
+    return [];
+  }
+
+  return orderedItems;
+}
+
+// the isUpKey function from the inquirer package is not used
+// because it detects k and p as custom keybindings that cause
+// the option selection to go up instead of writing the letters
+// in the search string
+function isUpKey(key: KeypressEvent): boolean {
+  return key.name === 'up';
+}
+
+// the isDownKey function from the inquirer package is not used
+// because it detects j and n as custom keybindings that cause
+// the option selection to go down instead of writing the letters
+// in the search string
+function isDownKey(key: KeypressEvent): boolean {
+  return key.name === 'down';
+}
+
+export const searchableCheckBox = createPrompt(
```

Comment:
I can't really tell that i'm able to search, until i start typing. Is there a good way to let the user know that it's searchable?

Reply:
Adding a hint to the help message should be okay I think. We already do the same for telling the user to use tabs to select and enter to complete the selection. Examples:

Before
![image](https://github.com/user-attachments/assets/13856f3b-a25e-4029-9f3c-70e1af4eeab9)

After
![image](https://github.com/user-attachments/assets/d929cc70-1673-4fab-a3d6-fd59cd515c38)